VladimirPopov_WebForms/templates/webforms/form/default.phtml

```diff
@@ -58,271 +79,294 @@ if ($this->getForm()->getIsActive() == \VladimirPopov\WebForms\Model\Form::STATU
-                                <?= ($fieldset['logic_visibility'] == \VladimirPopov\WebForms\Model\Logic::VISIBILITY_HIDDEN) ? 'display:none' : ''; ?>"
+                        <?php
+                        foreach ($fieldset['fields'] as $field) {
+                            // set temporary value
+                            $field->setData('result', $tmpResult);
+                            $field->setData('uid', $_uid);
+                            $field_css_type = 'type-' . str_replace('/', '-', $field->getType());
+                            if ($field->getType() == 'subscribe') $field_css_type .= ' choice';
+                            $aria_labelledby = "aria-labelledby=\"label_field" . $this->getFieldUid($field->getId()) . "\"";
+                            if (in_array($field->getType(), array("html", "subscribe")) || $field->getHideLabel())
+                                $aria_labelledby = "";
+                            $label_for = "for=\"field" . $this->getFieldUid($field->getId()) . "\"";
+                            if (in_array($field->getType(), array("select/radio", "select/checkbox"))) {
+                                $label_for = "";
+                            }
+
+                            $aria_describedby = "";
+                            if ($field->getComment()) $aria_describedby = "aria-describedby=\"comment_field" . $this->getFieldUid($field->getId()) . "\"";
+                        ?>
+                            <div
+                                id="field_<?= $block->escapeHtmlAttr($this->getFieldUid($field->getId())) ?>"
                                 role="group"
-                                <?php if ($fieldset_id): ?>
-                                    aria-labelledby="legend_<?= $block->escapeHtmlAttr($_uid) . $block->escapeHtmlAttr($fieldset_id) ?>"
+                                class="
+                                    field
+                                    <?= $block->escapeHtmlAttr($field->getCssClassContainer()) ?>
+                                    <?= $block->escapeHtmlAttr($field_css_type) ?>
+                                    webforms-fields-<?= $field->getId() ?>
+                                    <?php if ($field->getCode()): ?>
+                                        <?= 'webforms-fields-' . $block->escapeHtmlAttr($field->getCode()); ?>
+                                    <?php endif; ?>
+                                    <?php if ($field->getRequired()): ?>
+                                        required
+                                    <?php endif; ?>
+                                "
+                                <?= $aria_labelledby ?>
+                                <?= $aria_describedby ?>
+                                <?php if ($field->getData('logic_visibility') == \VladimirPopov\WebForms\Model\Logic::VISIBILITY_HIDDEN): ?>
+                                    <?= 'style="display:none"'; ?>
```

Comment:
can we add atomic class `display-none` instead of inline styles?

Reply:
yeah I think we can